lib/Doctrine/DBAL/Types/TypeRegistry.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Types;
+
+use Doctrine\DBAL\DBALException;
+use function array_search;
+use function in_array;
+
+/**
+ * @internal
+ */
+final class TypeRegistry
+{
+    /** @var Type[] Map of type names and their corresponding objects. */
+    private $instances = [];
+
+    /**
+     * Factory method to create type instances.
+     * Type instances are implemented as flyweights.
+     *
+     * @throws DBALException
+     */
+    public function get(string $name) : Type
+    {
+        if (! isset($this->instances[$name])) {
+            throw DBALException::unknownColumnType($name);
+        }
+
+        return $this->instances[$name];
+    }
+
+    /**
+     * Find a name for the given type.
```

Comment:
On line 51, you use the 3rd person, and here the imperative, maybe be stick to either one?